beacon_chain/types/src/beacon_state.rs

```diff
@@ -55,6 +57,32 @@ impl BeaconState {
         // https://github.com/sigp/lighthouse/issues/70
         Hash256::from(&canonical_hash(&ssz_encode(self))[..])
     }
+
+    /// Returns the `ShardCommittee` for the `slot`.
+    pub fn get_shard_committees_at_slot(
+        &self,
+        slot: u64,
+        epoch_length: u64,
+    ) -> Option<&Vec<ShardCommittee>> {
```

Comment:
re: `Vec<T>` or `&[T]` here, given that we maintain the committees as `Vec<T>` already, it is less computation to just provide a `&Vec<_>`. To get `&[_]`, we have to call `as_slice` on the target `Vec` which introduces another function call (and depending on how we get there, some pattern matching given that we are inside the `Option` context). It is possible that this gets optimized away by the compiler but I don't think we are at the point to be doing that kind of analysis wrt node performance.

i would elect to leave it as is for now, what do you think?

Reply:
yep @paulhauner s/shard_committee/crosslink_committee/

although i just checked the latest and the function has changed considerably bc we now compute the shuffling on-demand, rather than per-epoch and saving that to the state.

i like this in the spec, although it invalidates this function here...